### ISSUE TITLE
GitHub Actions Caching

### ISSUE THREAD

**Issue by metaskills** [customink/lamby#109]
So @vlaaaaaaad made this tweet how GitHub Actions now has great support for caching (https://twitter.com/iamvlaaaaaaad/status/1421529670353408001)

For us this is not likely a two line change since we use Docker in Docker. However, I wonder if we can share the right directories and take advantage of this?

**Comment by metaskills** [comment]
Useful Resources:

* [SAM CLI Issue Mentioning Buildx](https://github.com/aws/aws-sam-cli/issues/2951)
* [SAM Build Docs. Mentions Caching](https://docs.aws.amazon.com/serverless-application-model/latest/developerguide/sam-cli-command-reference-sam-build.html)
* [SAM Building Applications. Docker Metadata](https://docs.aws.amazon.com/serverless-application-model/latest/developerguide/serverless-sam-cli-using-build.html)

**Comment by metaskills** [comment]
Not sure this will for us without having to not use `sam build` and move a lot of things around to accommodate that since this action (https://github.com/docker/build-push-action) runs docker from the action and expects to build and push.

**Comment by synth** [comment]
Would love this. The majority of the build time is getting the ruby image. 

That said, it seems the new school way AWS wants us to do this with `sam build` is using [AWS SAM Accelerate](https://docs.aws.amazon.com/serverless-application-model/latest/developerguide/accelerate.html)

**Comment by metaskills** [comment]
Right, but I thought that was for Zip vs. Image format. Do you know? I have not looked.

**Comment by synth** [comment]
I'm not too sure. The documentation is sparse. The tutorial uses zip not image. I wonder if we can simply add the `--cached` to `sam build` if that would work. I can try later next week. 

References I found: 
* https://aws.amazon.com/blogs/compute/accelerating-serverless-development-with-aws-sam-accelerate/
* https://stackoverflow.com/questions/71638085/is-there-a-way-to-incrementally-build-container-images-with-aws-sam
* https://github.com/aws/aws-sam-cli/issues/805
* https://github.com/aws/aws-sam-cli/issues/3677

**Comment by synth** [comment]
Ah-may-zing! Thank you!
